packages/react-components/react-positioning/src/middleware/matchTargetSize.test.ts

```diff
@@ -1,4 +1,4 @@
-import type { MiddlewareArguments } from '@floating-ui/dom';
+import type { MiddlewareState } from '@floating-ui/dom';
```

Comment:
The type-check fails for this package because the `MiddlewareArguments` type is deprecated. Initially, I modified only the package.json scripts and tsconfig.json, and this package is among those affected, resulting in CI failure.

Reply:
discussed offline

- we gonna undo these changes within the package
- we install `@floating-ui/dom@1.2.0` to monorepo root (which should have been done when introducing this dependency to the package) in order to be compliant with our single version policy and setting the 3rd party packages we use and ship to production. this will keep the version on 1.2.0 so no changes will be needed in production code